openff/toolkit/utils/rdkit_wrapper.py

```diff
@@ -2298,137 +2299,225 @@ def _detect_undefined_stereo(cls, rdmol, err_msg_prefix="", raise_warning=False)
                 raise UndefinedStereochemistryError(msg)
 
     @staticmethod
-    def _flip_rdbond_direction(rdbond, paired_rdbonds):
-        """Flip the rdbond and all those paired to it.
+    def _constrain_end_directions(
+        *values: int, bond_indices: List[int], flip_direction: Dict[int, bool]
+    ) -> bool:
+        """A constraint applied when mapping global E/Z stereochemistry into local RDKit
+        bond directions that ensures that the 'left' bonds point in opposite directions
+        (i.e. one has to be up and one has to be down) and likewise for the 'right' bonds
+        """
+        # Account for bond "direction" using flip_directions dict, see more thorough comment
+        # in _constrain_rank for details.
+        unique_values = [
+            (value if not flip_direction[i] else 1 - value)
+            for i, value in zip(bond_indices, values)
+        ]
+        unique_values_set = set(unique_values)
+        print(unique_values_set)
```

Comment:
```suggestion
```

Is there a reason to use a list literal over a set literal? If you keep the swap I'd recommend re-naming `unique_values` to `bond_directions` as the values *won't* be unique until the list is cast to a set.

Reply:
I missed the curly braces the first few times I read this and I spent several minutes puzzling over how the logic in these two methods could work before realizing that equality was being checked by converting to a set. Since I have a terrible memory I know that I'll spend these minutes again in the future if I don't explicitly draw attention to the fact that this set conversion is happening.